src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1943,8 +1943,54 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
     if(symbol_table.symbols.find(identifier)==symbol_table.symbols.end())
     {
       // This is an undeclared function.
+
+      // Is it the polymorphic typed_target function ?
+      if(identifier == CPROVER_PREFIX "typed_target")
+      {
+        if(expr.arguments().size() != 1)
```

Comment:
The remainder of the code in this function sets out a poor example, but: could the body of this `if` be factored out into a function of its own?

Reply:
thanks, will do.